wa/workloads/aitutu/uiauto/app/src/main/java/com/arm/wa/uiauto/aitutu/UiAutomation.java

```diff
@@ -0,0 +1,147 @@
+import android.support.test.uiautomator.UiScrollable;
+import android.view.KeyEvent;
+import android.util.Log;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "UXPERF";
+
+    @Test
+    public void setup() throws Exception {
+        acceptPermissions();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void extractResults() throws Exception {
+        getScores();
+    }
+
+    public void acceptPermissions() throws Exception {
+        UiSelector selector = new UiSelector();
+        UiObject perm1 = mDevice.findObject(selector.textContains("1 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm2 = mDevice.findObject(selector.textContains("2 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm3 = mDevice.findObject(selector.textContains("3 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm4 = mDevice.findObject(selector.textContains("4 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject allow = mDevice.findObject(selector.textContains("ALLOW")
+                                             .className("android.widget.Button"));
+        if (perm1.exists()) {
+            allow.click();
+        }
+        if (perm2.exists()) {
+            allow.click();
+        }
+        if (perm3.exists()) {
+            allow.click();
+        }
+        if (perm4.exists()) {
+            allow.click();
+        }
+
+        UiObject next = mDevice.findObject(selector.textContains("NEXT")
+                                             .className("android.widget.Button"));
+        waitObject(next);
+        next.click();
+
+        UiObject cancel = mDevice.findObject(selector.textContains("CANCEL")
+                                             .className("android.widget.Button"));
+        waitObject(cancel);
+        cancel.click();
+    }
+
+    public void runBenchmark() throws Exception {
+        UiSelector selector = new UiSelector();
+        //Start the tests
+        UiObject start = mDevice.findObject(selector.textContains("Start Testing")
+                                                     .className("android.widget.TextView"));
+        waitObject(start);
+        start.click();
+
+        //Wait for the tests to complete
+        UiObject complete =
+            mDevice.findObject(selector.text("TEST AGAIN")
+                .className("android.widget.Button"));
+        complete.waitForExists(1200000);
+
+    }
+
+    public void getScores() throws Exception {
```

Comment:
Would it make sense to also extract which AIModel was used for the tests for better comparisons?

Reply:
Do you mean extracting the text element? Image Classification, Object Detection etc? 

If so I don't see the benefit, the scores are pulled to an element named either Image or Object.

Reply:
No sorry, I meant more "TFLite" vs "SNPE" etc.

Reply:
I'm not sure where these are specified in the application?

Reply:
On the results screen next to the score for each section it also lists the model that was used during the benchmark, for example here it has used the "TFLite" model. 
![aitutu](https://user-images.githubusercontent.com/22317974/54619769-0d79a280-4a5d-11e9-8c2b-17acfa853d0c.png)

Reply:
Completely missed that. Have now included.